packages/js/components/src/experimental-tree-control/stories/index.tsx

```diff
@@ -36,6 +36,36 @@ export const SimpleTree: React.FC = () => {
 	);
 };
 
+function isItemExpanded( item: LinkedTree, text: string ) {
+	if ( ! text || ! item.children?.length ) return false;
+	return item.children.some( ( child ) => {
+		if ( new RegExp( text, 'ig' ).test( child.data.label ) ) {
+			return true;
+		}
+		return isItemExpanded( child, text );
+	} );
+}
+
+export const ExpandOnSearch: React.FC = () => {
+	const [ text, setText ] = useState( '' );
+
+	return (
+		<>
+			<TextControl value={ text } onChange={ setText } />
+			<BaseControl label="Expand on search" id="expand-on-search">
+				<TreeControl
+					id="expand-on-search"
+					items={ listItems }
+					isItemExpanded={ useCallback(
```

Comment:
I think that `useCallback` is useful here. Otherwise, every time the parent of `TreeControl` was re-rendered, `TreeControl` would be re-rendered because the function passed into `isItemExpanded` would be re-created.

The function passed in is doing recursive checking of children items to see if an item should be expanded, so it seems potentially expensive if there is a large tree.

In this particular story, I don't think it really makes any noticeable difference though.

I could do either way, either keeping `useCallback` or getting rid of it for simplicity in this particular example.

Reply:
Currently in this example, the only time this should be re-rendered is when `text` is changed, right?  We would also want to re-render in that case and it does invalidate the memoization on the callback.

I think for this story (and most) the performance changes are going to be negligible either way, but since this decreases code readability and these stories act as some of the primary example for 3PDs, erring on the side of simplicity is best IMO.

Reply:
Agreed. Simplicity is the way to go. I'll update this.